cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -146,19 +146,25 @@ download_binary_and_run_installer() {
     assert_nz "$_true_arch" "arch"
     local _cur_arch="$_true_arch"
 
-    # Lookup what archives support this platform
+
+    # look up what archives support this platform
     local _artifact_name
     _artifact_name="$(select_archive_for_arch "$_true_arch")" || return 1
     local _bins
     local _zip_ext
     local _arch
+    local _checksum_style
+    local _checksum_value
 
-    # try each archive, checking runtime conditions like libc versions
-    # accepting the first one that matches, as it's the best match
+    # destructure selected archive info into locals
     case "$_artifact_name" in {% for archive in platform_support.archives %}
         "{{ archive.id }}")
             _arch="{{ archive.target_triple }}"
             _zip_ext="{{ archive.zip_style }}"
+            {%- if archive.checksum %}
```

Comment:
Huge gotcha here: `archive.checksum` is an index to an artifact struct, not the struct itself. Because jinja2 was originally designed for Python, `0` is falsy - and so is `Some(0)`. This check has the chance to return false for certain checksums. See `{%- if archive.updater != None %}` below in this same case statement for another place we had to work around this.

We also need to make sure `_checksum_style` and `_checksum_value` get initialized to somethign since we do `set -u` at the top of the file - `local _checksum_style` isn't enough to define it, so referencing it later will error out. (You'll see that happen in CI on this PR.)